chaste_codegen/_linearity_check.py

```diff
@@ -13,6 +13,14 @@ class KINDS(Enum):
 
 
 def check_expr(expr, state_var, membrane_voltage_var, state_vars):
+    """Check the kind of expression given (NONE, LINEAR or NONLINEAR)
```

Comment:
If you're going to add docstrings, I'd suggest making them more than 1-liners. There's quite a bit written about the algorithm, context etc which you could copy from pycml to help future readers.

Reply:
The explanation is already there but I've tried to expand it a bit further.